explainability-service/src/main/java/org/kie/trustyai/service/prometheus/PrometheusPublisher.java

```diff
@@ -38,37 +41,23 @@ private void createOrUpdateGauge(String name, Iterable<Tag> tags, UUID id) {
 
     }
 
-    private Iterable<Tag> generateTags(String modelName, UUID id, ReconciledMetricRequest request) {
-        return Tags.of(
-                Tag.of("model", modelName),
-                Tag.of("outcome", request.getOutcomeName()),
-                Tag.of("favorable_value", request.getFavorableOutcome().toString()),
-                Tag.of("protected", request.getProtectedAttribute()),
-                Tag.of("privileged", request.getPrivilegedAttribute().toString()),
-                Tag.of("unprivileged", request.getUnprivilegedAttribute().toString()),
-                Tag.of("batch_size", String.valueOf(request.getBatchSize())),
-                Tag.of("request", id.toString()));
+    private Iterable<Tag> generateTags(String modelName, UUID id, BaseMetricRequest request) {
+        List<Tag> tags = request.retrieveTags().entrySet().stream()
+                .map(e -> Tag.of(e.getKey(), e.getValue()))
+                .collect(Collectors.toList());
+        tags.add(Tag.of("request", id.toString()));
+        tags.add(Tag.of("model", modelName));
+        return Tags.of(tags);
     }
 
-    public void gaugeSPD(ReconciledMetricRequest request, String modelName, UUID id, double value) {
+    public void gauge(@ValidReconciledMetricRequest BaseMetricRequest request, String modelName, UUID id, double value) {
 
         values.put(id, new AtomicDouble(value));
 
         final Iterable<Tag> tags = generateTags(modelName, id, request);
 
-        createOrUpdateGauge("trustyai_spd", tags, id);
+        createOrUpdateGauge("trustyai_" + request.getMetricName().toLowerCase(), tags, id);
```

Comment:
We might want to do a more "advanced" escaping strategy (i.e. check for ` ` or any other symbol) to match the expected format ( https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels ).
If you agree feel free to open a follow up ticket.
(as alternative we can validate the name of the metrics and fail at startup time if it is not valid as format)

Reply:
Yep, I agree, will do. https://github.com/trustyai-explainability/trustyai-explainability/issues/297